google/cloud/spanner_dbapi/connection.py

```diff
@@ -231,6 +268,19 @@ def transaction_checkout(self):
 
             return self._transaction
 
+    def snapshot_checkout(self):
+        """Get a Cloud Spanner snapshot.
+
+        Initiate a new multi-use snapshot, if there is no snapshot in
+        this connection yet. Return the existing one otherwise.
+
+        :rtype: :class:`google.cloud.spanner_v1.snapshot.Snapshot`
+        :returns: A Cloud Spanner snapshot object, ready to use.
+        """
+        if not self._snapshot:
+            self._snapshot = Snapshot(self._session_checkout(), multi_use=True)
+        return self._snapshot
```

Comment:
I think it would be good in that case to add a check at the beginning of the method that actually checks that the state of the connection is `read_only==True && autocommit==False`. That prevents any accidental errors in the future when making changes.

Reply:
This was marked as resolved but no changes were made. Please address.

Reply:
Done